conans/client/installer.py

```diff
@@ -385,14 +385,18 @@ def _get_remote_package(self, conan_file, package_reference, output, package_fol
         if not os.path.exists(package_folder):
             self._out.info("Retrieving package %s" % package_reference.package_id)
 
-        if self._remote_proxy.get_package(package_reference,
-                                          short_paths=conan_file.short_paths):
-            _handle_system_requirements(conan_file, package_reference,
-                                        self._client_cache, output)
-            if get_env("CONAN_READ_ONLY_CACHE", False):
-                make_read_only(package_folder)
-            return True
-
+        try:
+            if self._remote_proxy.get_package(package_reference,
+                                              short_paths=conan_file.short_paths):
+                _handle_system_requirements(conan_file, package_reference,
+                                            self._client_cache, output)
+                if get_env("CONAN_READ_ONLY_CACHE", False):
+                    make_read_only(package_folder)
+                return True
+        except Exception:
+            if os.path.exists(package_folder):
+                rmdir(package_folder)
```

Comment:
Maybe something like:
```python
        try:
            rmdir(self.build_folder)
            rmdir(self.package_folder)
        except OSError as e:
            raise ConanException("%s\n\nCouldn't remove folder, might be busy or open\n"
                                 "Close any app using it, and retry" % str(e))
```
We have it in other parts of the codebase.
The thing is that this rmdir can eventually fail, leaving an ugly and not easy to understand trace.
But we don't follow it thoroughly in our code base, so this is already and improvement, so I am fine with it as-is, up to you.